### ISSUE TITLE
NextJS 11.1.4 - conflicting extension of react 'module' declaration

### ISSUE THREAD

**Issue by rr-justin-hanselman** [vercel/next.js#34022]
### Run `next info` (available from version 12.0.8 and up)

Working on a library - this does not easily run the next command since there's not next configs.

typescript - 4.4.2
nextJs - 11.1.4
react - 17.0.2

### What version of Next.js are you using?

11.1.4

### What version of Node.js are you using?

12.22.9

### What browser are you using?

N/A

### What operating system are you using?

macOS

### How are you deploying your application?

All next options

### Describe the Bug

When building typescript libraries that extend NextJS 11.1.4 and use react 17.0.2, I receive the following error:

```
node_modules/next/types/index.d.ts:46:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'loading' must be of type '"eager" | "lazy" | undefined', but here has type '"auto" | "eager" | "lazy" | undefined'.

46     loading?: 'auto' | 'eager' | 'lazy'
       ~~~~~~~

  ../../node_modules/@types/react/index.d.ts:2156:9
    2156         loading?: "eager" | "lazy" | undefined;
                 ~~~~~~~
    'loading' was also declared here.
```

The cause of this is that NextJS bundles extensions to the react type definitions BUT it conflicts with type-checking.  For now, I will be using skipLibCheck, but that is a nuclear option, since it skips all lib checking.  

This issue was [previously closed](https://github.com/vercel/next.js/issues/29071) with the mention that it was fixed in the canary last year.  I'm filing this bug since it doesn't look updated.  

Could we release a patched version that doesn't conflict with react types?

### Expected Behavior

Typescript transpiles everything correctly, and uses the appropriate type declarations from react instead of NextJS.

### To Reproduce

Create a typescript project that makes use of both the 'next' package and the react package.

```typescript
// index.ts
import React, { useState } from 'react';
import type { Redirect } from 'next';

export const MyComp : React.FC = ({ children }) => {
    const redirect = useState<Redirect>({
      statusCode: 301,
      destination: 'some-destination',
    });

    return (<div>This component is using both next and react packages</div>);
};
```

Once you've created the project:

```bash
tsc
```

You should get the same failure as above (as long as skipLibCheck does not equal true in your tsconfig.json).

**Comment by rr-justin-hanselman** [comment]
As a temporary solution that doesn't make me skipLibChecks on everything, I am using ignore-bad-ts.  I've added it as a yarn 3 plugin in my project, but you could easily call it via cli

```
npx ignore-bad-ts "node_modules/next/types/index.d.ts"
```

IMPORTANT: the package just adds ts-nocheck to the top of files that match the glob.  So you will want to clean your packages if you install a fixed version of NextJS and remove the ignore-bad-ts calls.  I prefer this though, so that I still get my typechecking for everything else except next.

**Comment by balazsorban44** [comment]
I don't think we are going to backport patches like this. Upgrading to Next 12 solves the issue, with the one caveat that if you want `"skipLibCheck": false`, you might need to install `@types/react-dom` as a dev dependency as well.

**Comment by rr-justin-hanselman** [comment]
Just to be clear, changing a type definition for a still-supported version of NextJS is considered a breaking change.  I'm not sure what all the NextJS eco-system is reliant on, but that seems a little drastic.

If this is the final resolution, for anyone who comes on this and doesn't want to disable all typechecking, please see my above workaround and keep an eye out for new NextJS releases that might implicitly fix this. :D

**Comment by balazsorban44** [comment]
I might have been unclear. This has been fixed already in Next.js 12 and upwards, no need for PR or any further fix.

What I meant was that it is fairly easy to upgrade between major releases of Next.js, since the breaking changes are kept at a very low level, making it a very stable framework. I did not say that a type fix would be a breaking change, only that we are not implementing patches/fixes/features on older versions of Next.js unless it addresses a security issue.
